packages/tools/client-debugger/client-debugger-view/src/test/widgets/EmojiButton.tsx

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { TooltipHost } from "@fluentui/react";
+import { Button, useId } from "@fluentui/react-components";
+import React from "react";
+
+import { SharedCell } from "@fluidframework/cell";
+
+/**
+ * {@link EmojiButton} input props.
+ */
+export interface EmojiButtonProps {
+	emojiCell: SharedCell<boolean>;
+}
+
+/**
+ * Simple button that displays either a smily or frouny emoji.
+ * Pressing the button toggles between the two.
+ *
+ * State is shared via the provided `SharedCell`.
+ */
+export function EmojiButton(props: EmojiButtonProps): React.ReactElement {
+	const { emojiCell } = props;
+
+	// undefined => No expression
+	// false => frouny
+	// true => smily
+	const [isSmily, setIsSmily] = React.useState<boolean | undefined>(emojiCell.get());
+
+	React.useEffect(() => {
+		function updateState(): void {
+			setIsSmily(emojiCell.get());
+		}
+
+		emojiCell.on("valueChanged", updateState);
+		emojiCell.on("delete", updateState);
+
+		return (): void => {
+			emojiCell.off("valueChanged", updateState);
+			emojiCell.on("delete", updateState);
+		};
+	}, [emojiCell, isSmily, setIsSmily]);
+
+	const buttonTooltipId = useId("decrement-counter-button");
```

Comment:
This sounds like it might conflict with the buttons of the counter widget...?

Reply:
Yeah, this is a bad copy-paste. Will update.